.github/workflows/changelogs/changelogs.py

```diff
@@ -0,0 +1,639 @@
+        the 'rel-eng/packages' directory. Each file in this directory defines
+        the name of a package (the name of the file), the current version of the
+        package, and its base path.
+        """
+
+        packages_dir = os.path.join(self.spacewalk_root, "rel-eng/packages")
+        pkg_idx = {}
+
+        try:
+            pkg_names = os.listdir(packages_dir)
+            logging.debug(f"Found {len(pkg_names)} package(s) in 'rel-eng/packages'")
+        except FileNotFoundError:
+            raise Exception(f"Not an Uyuni repository. Consider using '--spacewalk-dir' option.")
+
+        for pkg_name in pkg_names:
+            if pkg_name.startswith('.'):
+                # Skip hidden files in rel-eng/packages
+                continue
+            # Extract the package path from the file:
+            # Each file contains the package version and the
+            # package path, separated by a space character
+            pkg_path = linecache.getline(os.path.join(packages_dir, pkg_name), 1).rstrip().split(maxsplit=1)[1]
+            logging.debug(f"Package {pkg_name} is in path {pkg_path}")
+
+            # Get the list of modified files and changelog files for the package
+            modified_files = self.get_modified_files_for_pkg(pkg_path, pkg_name, files)
+            if modified_files["files"] or modified_files["changes"]:
+                pkg_idx[pkg_name] = modified_files
+
+        return pkg_idx
+
+    def extract_trackers(self, text: str) -> dict[str, list[tuple[str, str]]]:
+        """Extract all the mentioned trackers in a body of text
+
+        The trackers to be extracted are defined in the 'regex' object.
+
+        The method returns a dictionary of trackers in the following format:
+
+            ["kind"] -> [(full_tracker_id, num_tracker_id)]
+
+            Example of a tuple: ("bsc#1234567", "1234567")
+        """
+
+        trackers = {}
+        for kind, regex in self.regex.trackers.items():
+            trackers_of_kind = []
+            for match in re.finditer(regex, text):
+                # Match groups are defined by contract in the tracker file
+                # Group 1 should be the numeric ID of the tracker
+                trackers_of_kind.append(match.group(0, 1))
+
+            # Gather the unique trackers
+            trackers[kind] = list(set(trackers_of_kind))
+
+        return trackers
+
+    def get_pr_trackers(self, git_repo: str, pr_number: int) -> dict[str, list[tuple[str, str]]]:
+        """Get all the trackers mentioned in a PR
+
+        The trackers are extracted from the PR title and the commit messages.
+        """
+
+        assert pr_number
+        assert git_repo
+
+        logging.info(f"Requesting information for PR#{pr_number} at '{git_repo}'")
+        stream = os.popen(f'gh pr view -R {git_repo} {pr_number} --json title,commits -q ".title, .commits[].messageHeadline, .commits[].messageBody | select(length > 0)"')
```

Comment:
Is the dependency on `gh` documented anywhere? Would be useful for devs who want to play with the script locally.

Reply:
I added the following to the README:

> **Note:** To run the script, you need to have the `gh` (GitHub CLI tool) and the `python-bugzilla-3.2.0` library installed.